pkg/ui/frontend/src/components/HP/Create/HPParameters.jsx

```diff
@@ -0,0 +1,163 @@
+        let  tempParam = {}
+        tempParam.name = param.name
+        tempParam.parametertype = param.parameterType
+        tempParam.feasible = {}
+        if (param.feasible === "list") {
+            tempParam.feasible.list = param.list.map((param, i) => param.value)
+        } else {
+            tempParam.feasible.min = param.min
+            tempParam.feasible.max = param.max
+        }
+        destination.push(tempParam)
+    })
+}
+const SectionInTypography = (name, classes) => {
+    return (
+        <div className={classes.section}>
+            <Grid container>
+                <Grid item xs={12} sm={12}>
+                    <Typography variant="h6">
+                        {name}
+                    </Typography>
+                <hr />
+                </Grid>
+            </Grid>
+        </div>
+    )
+}
+
+// probably get render into a function
+
+const HPParameters = (props) => {
+    const submitJob = () => {
+        let data = {}
+        data.metadata = {}
+        deCapitalizeFirstLetterAndAppend(props.commonParametersMetadata, data.metadata)
+        data.spec = {}
+        deCapitalizeFirstLetterAndAppend(props.commonParametersSpec, data.spec)
+
+        data.spec.parameterconfigs = []
+        addParameterConfig(props.parameterConfig, data.spec.parameterconfigs)
+        data.spec.workerSpec = {
+            goTemplate: {
+                templatePath: props.worker,
+            }
+        }
+        data.spec.metricsnames = props.metricsName.map((metrics, i) => metrics.value)
+        data.spec.suggestionSpec = {}
+        data.spec.suggestionSpec.requestNumber = (!isNaN(Number(props.requestNumber)) ? Number(props.requestNumber) : 1)
+        data.spec.suggestionSpec.suggestionAlgorithm = props.suggestionAlgorithm
+        data.spec.suggestionSpec.suggestionParameters = []
+        addSuggestionParameters(props.suggestionParameters, data.spec.suggestionSpec.suggestionParameters)
+        props.submitHPJob(data)
+    }
+
+    const { classes } = props;
+
+    return (
+            <div className={classes.root}>
+                {/* Common Metadata */}
+                {SectionInTypography("Metadata", classes)}
+                <br />
+                <CommonParametersMeta />
+                {SectionInTypography("Spec", classes)}
+                <CommonParametersSpec />
+                {SectionInTypography("Parameters Config", classes)}
+                <ParameterConfig />
+                {SectionInTypography("Worker Spec", classes)}
+                <WorkerSpecParam />
+                {SectionInTypography("Suggestion Parameters", classes)} 
+                
+                <SuggestionSpec />
+                <div className={classes.submit}>
+                    <Button variant="contained" color={"primary"} className={classes.button} onClick={submitJob}>
+                        Deploy
+                    </Button>
+                </div>                
+            </div>
+    )
+}
```

Comment:
may be useful to add type checking on the props to components https://reactjs.org/docs/typechecking-with-proptypes.html

Reply:
Nice call, doing it rn.